### DISCUSSION TITLE
Support for Python QML Elements

### DISCUSSION THREAD

**Discussion by machinekoder** [seanwu1105/vscode-qt-for-python#312] [Ideas]
It's possible to generate `plugins.qmltypes` file for Python-based QML elements using an undocumented feature.
The `pyside6-metaobjectdump` tool can be used to generate a `moc` file which can be used by `pyside6-qmltyperegistrar` to generate the `qmltypes` file, which is picked up by `qmllint` and other Qt tools when added to the `qmldir` file as `typeinfo` entry.

What would be the best way to implement this feature as part of the VSCode plugin?

**Comment by seanwu1105** [comment]
The extension can trigger pyside6-metaobjectdump on a Python-based QML element updated. The same idea could be applied to other Pyside tools but since I don't have experience with these tools, a PR or some minimal use cases description could be really helpful for me.
